server/server.go

```diff
@@ -71,15 +73,21 @@ func New(c *Config, log internal.Logger) *Server {
 		c = &conf
 	}
 	s := &Server{
-		c:       *c,
-		log:     log,
-		players: make(chan *player.Player),
-		world:   world.New(log, c.World.SimulationDistance),
-		p:       make(map[uuid.UUID]*player.Player),
-		name:    *atomic.NewString(c.Server.Name),
+		c:              *c,
+		log:            log,
+		players:        make(chan *player.Player),
+		world:          world.New(log, c.World.SimulationDistance),
+		p:              make(map[uuid.UUID]*player.Player),
+		name:           *atomic.NewString(c.Server.Name),
+		playerProvider: player.NopProvider{},
 	}
 	s.JoinMessage(c.Server.JoinMessage)
 	s.QuitMessage(c.Server.QuitMessage)
+	if p, err := provider.NewDBProvider("players"); err == nil {
+		s.PlayerProvider(p)
+	} else {
+		log.Errorf("Error starting player provider: %v", err)
```

Comment:
Perhaps you should return this error or panic? I imagine player data saving would be a pretty critical component

Reply:
Okie! I added a panic and I also added an option in the config to disable the default provider, that way players who dont want it can explicitly say so, and it can prevent a database being created for servers that don't use it.